### PR TITLE
Add more logging for delete requests

### PR DESCRIPTION
**Pull request by MichelHollands**
**What this PR does / why we need it**:

Add more logging for the delete requests.

**Checklist**
- [ ] Documentation added
- [ ] Tests updated
- [ ] Is this an important fix or new feature? Add an entry in the `CHANGELOG.md`.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

Signed-off-by: Michel Hollands <michel.hollands@grafana.com>

### CODE REVIEW

File: pkg/storage/stores/shipper/compactor/deletion/delete_requests_manager.go [hunk 1]
```diff
@@ -163,9 +168,19 @@ func (d *DeleteRequestsManager) Expired(ref retention.ChunkEntry, _ model.Time)
 
 		d.chunkIntervalsToRetain = rebuiltIntervals
 		if len(d.chunkIntervalsToRetain) == 0 {
+			level.Info(util_log.Logger).Log(
+				"msg", "no chunks to retain for delete request",
```
Comment (MasslessParticle) [inline pkg/storage/stores/shipper/compactor/deletion/delete_requests_manager.go:172]: This means we delete the whole chunk, right? Would it be good to add that to the log?

Comment (MichelHollands) [reply pkg/storage/stores/shipper/compactor/deletion/delete_requests_manager.go]: Good idea. I've added the ChunkID as well.

File: pkg/storage/stores/shipper/compactor/deletion/delete_request.go [hunk 2]
```diff
@@ -97,13 +97,30 @@ func (d *DeleteRequest) IsDeleted(entry retention.ChunkEntry) (bool, []retention
 		return false, nil
 	}
 
+	level.Debug(util_log.Logger).Log(
+		"msg", "starting filter function",
+		"delete_request_id", d.RequestID,
+		"userID", d.UserID, "labels",
```
Comment (dannykopping) [inline pkg/storage/stores/shipper/compactor/deletion/delete_request.go:103]: `<nit>`

```suggestion
		"user", d.UserID, "labels",
```

`</nit>`

We have far more logs with the key "user" than "userID" and it'd be great to keep some consistency here.

```bash
loki [main●●] $ grep '"userID"' -rn pkg | grep log | wc -l
8
loki [main●●] $ grep '"user"' -rn pkg | grep log | wc -l  
43
```

File: pkg/storage/stores/shipper/compactor/deletion/delete_request.go [hunk 3]
```diff
@@ -97,13 +97,30 @@ func (d *DeleteRequest) IsDeleted(entry retention.ChunkEntry) (bool, []retention
 		return false, nil
 	}
 
+	level.Debug(util_log.Logger).Log(
+		"msg", "starting filter function",
+		"delete_request_id", d.RequestID,
```
Comment (vlad-diachenko) [inline pkg/storage/stores/shipper/compactor/deletion/delete_request.go:102]: should we also use camel-case here ? I see that for `userID` we use camel-case

**Review by vlad-diachenko — APPROVED** [review]
LGTM

**Review by ssncferreira — APPROVED** [review]
LGTM

File: pkg/storage/stores/shipper/compactor/deletion/delete_request.go (hunk 2 above)
Comment (MichelHollands) [reply pkg/storage/stores/shipper/compactor/deletion/delete_request.go:103]: Good idea. This was changed manually on the 3 places where userID was used.

File: pkg/storage/stores/shipper/compactor/deletion/delete_request.go (hunk 3 above)
Comment (MichelHollands) [reply pkg/storage/stores/shipper/compactor/deletion/delete_request.go:102]: Due to a previous comment `userID` is now changed to `user`. Regarding camel-case or not, we seem to be somewhat inconsistent. There are a lot of examples of each style in the code.

Comment (dannykopping) [reply pkg/storage/stores/shipper/compactor/deletion/delete_request.go:102]: No, we generally use snake_case in our log keys

Comment (vlad-diachenko) [reply pkg/storage/stores/shipper/compactor/deletion/delete_request.go:102]: ok, great ;)

**Review by dannykopping — APPROVED** [review]
LGTM, thanks @MichelHollands

**Pull request merged**